README.rst

```diff
@@ -256,6 +256,57 @@ Options:
 :usage-type:		Specify the usage type for the filesystem. Only valid with mke2fs.
 			More details can be found in the mke2fs man-page.
 
+file
+****
+
+This represents a pre-existing image which will be used as-is. When a
+partition section references an image that is not defined elsewhere in
+the configuration file, a ``file`` rule is implicitly generated. It is
+up to the user to ensure that the image exists in the input directory,
+or to use an absolute path to the image.
+
+It is possible to add a ``file`` image explicitly, which allows one to
+provide ``genimage`` with some information about the image which can
+not be deduced automatically. Currently, one such option exists:
+
+:holes:			A list of ``"(<start>;<end>)"`` pairs specifying ranges of the
+			file that do not contain meaningful data, and which can therefore
+			be allowed to overlap other partitions or image metadata.
+
+For example::
+
+  image foo {
+	  hdimage {
+		  gpt = true
+		  gpt-location = 64K
+	  }
+
+	  partition bootloader {
+		  in-partition-table = false
+		  offset = 0
+		  image = "/path/to/bootloader.img"
+	  }
+
+	  partition rootfs {
+		  offset = 1M
+		  image = "rootfs.ext4"
+	  }
+  }
+
+  image /path/to/bootloader.img {
+	  file {
+		  holes = {"(440; 17K)", "(64K; 80K)"}
```

Comment:
I'd use `2k` instead of `17k` for the first hole. That's more realistic. That's the hole needed for DOS partition table and the GPT header.

Reply:
Wouldn't that be 1k? One section for MBR, another for GPT header? I chose 17k because that would then also cover the gpt array - which is rarely moved unless absolutely necessary (e.g. imx platforms with bootloader at 1k). So you could have a bootloader at offset 0 which is prepared to work with a standard GPT layout.

Reply:
Yes, 1k not 2k of course. And in your example you move the GPT table so I think that would be consistent.
By the way, on newer imx SoCs (e.g. imx8mq) you don't need to move the GPT table. Here the ROM code looks at 33k.